### PR TITLE
docs: add a typography guide

### PR DESCRIPTION
**Pull request by crisbeto**
Adds a guide that shows how to use the typography setup.

### CODE REVIEW

File: guides/typography.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+* `mat-{{component}}-typography` - Every Material component has a corresponding typography mixin,
+which allows for its styles to be overwritten (e.g. `mat-checkbox-typography`).
+
+For more details about the typography functions and default config, see the
+[source](https://github.com/angular/material2/blob/master/src/lib/core/typography/_typography.scss).
+
+
+```scss
+// Import the requied mixins.
+@import '~@angular/material/theming';
```
Comment (crisbeto) [inline guides/typography.md]: @jelbourn this includes all the necessary typography mixins, but I'm not sure whether the file name won't cause some confusion. We could potentially copy it over and call it `typography.scss`.

**Comment by crisbeto** [comment]
There we go, yours has a much better flow to it.

**Comment by jelbourn** [comment]
Someone else should review since it's not somewhat co-authored

File: guides/typography.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+descendant native elements.
+
+```html
+<!-- By default, Angular Material applies no global styles to native elements. -->
+<h1>This header is unstyled</h1>
+
+<!-- Applying the mat-tyography class adds styles for native elements. -->
+<section class="mat-typography">
+  <h1>This header will be styled</h1>
+  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nihil, molestias.</p>
```
Comment (andrewseguin) [inline guides/typography.md]: Why is the <p> element added in the comparison? Makes it look like its relevant to the point this section is making

File: guides/typography.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  $font-family: monospace,
+  $headline: mat-typography-level(32px, 48px, 700),
+  $body-1: mat-typography-level(16px, 24px, 500)
+);
+```
+
+As the above example demonstrates, a typography configuration is created by using the
+`mat-typography-config` function, which is given both the font-family and the set of typographic
+levels described earlier. Each typographic level is defined by the `mat-typography-level` function,
+which requires a font-size, line-height, and font-weight.
```
Comment (andrewseguin) [inline guides/typography.md]: Change to `font-size`, `line-height` and `font-weight` just like line 6

File: guides/typography.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+For more details about the typography functions and default config, see the
+[source](https://github.com/angular/material2/blob/master/src/lib/core/typography/_typography.scss).
+
+
+### Material typography in your custom CSS
+Angular Material includes typography utility mixins and functions that you can use to customize your
+own components:
+
+* `mat-font-size($config, $level)` - Gets the font-size, based on the provided config and level.
```
Comment (andrewseguin) [inline guides/typography.md]: `font-size`, `font-family`, etc. 

Apply the code styles?

File: guides/typography.md (hunk 2 above)
Comment (crisbeto) [inline guides/typography.md]: It's mostly to show that the paragraph would get styled as well, but I can see how it could cause confusion.

**Comment by crisbeto** [comment]
Feedback is addressed.

File: guides/typography.md (hunk 1 above)
Comment (amcdnl) [inline guides/typography.md]: I favor calling it `typography.scss` too.

**Pull request merged**